libs/platform/unit/platform_unit_test.cpp

```diff
@@ -634,13 +634,15 @@ TEST_CASE("serialize_map_test", "[platform]")
     }
 
     // Serialize.
-    ebpf_result_t result = ebpf_serialize_internal_map_info_array(
-        map_count, internal_map_info_array, buffer, buffer_length, &serialized_length, &required_length);
-    REQUIRE(result == EBPF_INSUFFICIENT_BUFFER);
+    REQUIRE(
+        ebpf_serialize_internal_map_info_array(
+            map_count, internal_map_info_array, buffer, buffer_length, &serialized_length, &required_length) ==
+        EBPF_INSUFFICIENT_BUFFER);
 
-    buffer = static_cast<uint8_t*>(calloc(required_length, 1));
-    REQUIRE(buffer != nullptr);
-    if (!buffer) {
+    buffer = static_cast<uint8_t*>(ebpf_allocate(required_length));
+    // Required to deal with code analysis warning about buffer not being checked for null.
+    if (buffer == nullptr) {
+        REQUIRE(false);
```

Comment:
nit: can we change it back to `REQUIRE(buffer != nullptr)` ?

Reply:
Unfortunately not. The code analysis tool doesn't see REQUIRE(buffer != nullptr) as observing the buffer handle.